rating/src/main/java/com/yas/rating/service/RatingService.java

```diff
@@ -64,4 +65,15 @@ public RatingVm createRating(RatingPostVm ratingPostVm) {
         Rating savedRating = ratingRepository.saveAndFlush(rating);
         return RatingVm.fromModel(savedRating);
     }
+
+    public Double calculateAverageStar(Long productId){
+        List<Object[]> totalStarsAndRatings = ratingRepository.getTotalStarsAndTotalRatings(productId);
+        if(totalStarsAndRatings.get(0)[0] == null || totalStarsAndRatings.get(0)[1] == null)
```

Comment:
In my opinion, you should check if CollectionUtils.isEmpty(totalStarsAndRatings) then return 0.0 first